stores/channel_store.jsx

```diff
@@ -546,27 +549,28 @@ ChannelStore.dispatchToken = AppDispatcher.register((payload) => {
         });
         break;
 
-    case ActionTypes.RECEIVED_POST:
-        if (Constants.IGNORE_POST_TYPES.indexOf(action.post.type) !== -1) {
+    case ActionTypes.RECEIVED_POST: {
+        const {post, websocketMessageProps: data} = action;
+        const {dispatch, getState} = store;
+        if (shouldIgnorePost(post)) {
             return;
         }
 
-        if (action.post.user_id === UserStore.getCurrentId() && !isSystemMessage(action.post) && !isFromWebhook(action.post)) {
-            return;
+        let markAsRead = false;
+        if (post.user_id === UserStore.getCurrentId() && !isSystemMessage(post) && !isFromWebhook(post)) {
+            markAsRead = true;
+        } else if (action.post.channel_id === ChannelStore.getCurrentId() && window.isActive) {
+            markAsRead = true;
         }
 
-        var id = action.post.channel_id;
-        var teamId = action.websocketMessageProps ? action.websocketMessageProps.team_id : null;
-        var markRead = id === ChannelStore.getCurrentId() && window.isActive;
-
-        if (TeamStore.getCurrentId() === teamId || teamId === '') {
-            if (!markRead) {
-                ChannelStore.incrementMentionsIfNeeded(id, action.websocketMessageProps);
-            }
-            ChannelStore.incrementMessages(id, markRead);
+        if (markAsRead) {
+            markChannelAsRead(post.channel_id, null, false)(dispatch, getState);
```

Comment:
Can you change these around to `dispatch(markChannelAsRead(post.channel_id, null, false));`? I meant to bring this up to everyone because doing it this way makes it work regardless of if the action returns a function for an async action or if it returns a plain object like a regular redux action